File: Audexa/app/src/main/java/com/kreativekoala/audexa/data/model/Voice.kt
package com.kreativekoala.audexa.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * TTS Provider options
 */
enum class TTSProvider(val value: String, val displayName: String, val description: String) {
    OPENAI("openai", "OpenAI", "Fast, high-quality neural voices"),
    ELEVENLABS("elevenlabs", "ElevenLabs", "Ultra-realistic with voice cloning");

    companion object {
        fun fromValue(value: String): TTSProvider? {
            return entries.find { it.value == value }
        }
    }
}

/**
 * Unified voice representation from API
 */
@Serializable
data class Voice(
    val id: String,
    val name: String,
    val provider: String,
    val gender: String,
    val accent: String? = null,
    val description: String? = null,
    @SerialName("preview_url")
    val previewUrl: String? = null,
    val category: String? = null
) {
    val displayGender: String
        get() = gender.replaceFirstChar { it.uppercase() }

    val providerType: TTSProvider?
        get() = TTSProvider.fromValue(provider)

    val genderIcon: String
        get() = when (gender.lowercase()) {
            "male" -> "person"
            "female" -> "person"
            else -> "person_outline"
        }
}

/**
 * Voice pair recommendation for two-host podcasts
 */
@Serializable
data class VoicePair(
    val id: String,
    val name: String,
    val provider: String,
    val host1: Voice,
    val host2: Voice,
    val description: String
) {
    val providerType: TTSProvider?
        get() = TTSProvider.fromValue(provider)
}

/**
 * API response for voices endpoint
 */
@Serializable
data class VoicesResponse(
    val success: Boolean,
    val total: Int,
    val voices: List<Voice>,
    val grouped: Map<String, List<Voice>>? = null
)

/**
 * API response for voice pairs endpoint
 */
@Serializable
data class VoicePairsResponse(
    val success: Boolean,
    val pairs: List<VoicePair>
)

/**
 * API response for providers endpoint
 */
@Serializable
data class ProvidersResponse(
    val success: Boolean,
    val providers: List<ProviderInfo>
)

/**
 * Provider info from API
 */
@Serializable
data class ProviderInfo(
    val id: String,
    val name: String,
    val description: String,
    val available: Boolean,
    val configured: Boolean,
    @SerialName("voice_count")
    val voiceCount: Int,
    val features: List<String>
)

/**
 * Current voice configuration
 */
data class VoiceConfiguration(
    val provider: TTSProvider,
    val host1VoiceId: String,
    val host2VoiceId: String
) {
    companion object {
        val default = VoiceConfiguration(
            provider = TTSProvider.OPENAI,
            host1VoiceId = "openai:nova",
            host2VoiceId = "openai:onyx"
        )
    }
}

File: BriefCast/BriefCast/Models/Voice.swift
//
//  Voice.swift
//  BriefCast
//
//  Voice models for TTS provider selection
//

import Foundation

/// TTS Provider options
enum TTSProvider: String, Codable, CaseIterable {
    case openai = "openai"
    case elevenlabs = "elevenlabs"

    var displayName: String {
        switch self {
        case .openai: return "OpenAI"
        case .elevenlabs: return "ElevenLabs"
        }
    }

    var description: String {
        switch self {
        case .openai: return "Fast, high-quality neural voices"
        case .elevenlabs: return "Ultra-realistic with voice cloning"
        }
    }

    var icon: String {
        switch self {
        case .openai: return "waveform"
        case .elevenlabs: return "waveform.circle.fill"
        }
    }
}

/// Unified voice representation from API
struct Voice: Codable, Identifiable, Hashable {
    let id: String
    let name: String
    let provider: String
    let gender: String
    let accent: String?
    let description: String?
    let previewUrl: String?
    let category: String?

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case provider
        case gender
        case accent
        case description
        case previewUrl = "preview_url"
        case category
    }

    var displayGender: String {
        gender.capitalized
    }

    var providerType: TTSProvider? {
        TTSProvider(rawValue: provider)
    }

    var genderIcon: String {
        switch gender.lowercased() {
        case "male": return "person.fill"
        case "female": return "person.fill"
        default: return "person.fill.questionmark"
        }
    }
}

/// Voice pair recommendation for two-host podcasts
struct VoicePair: Codable, Identifiable {
    let id: String
    let name: String
    let provider: String
    let host1: Voice
    let host2: Voice
    let description: String

    var providerType: TTSProvider? {
        TTSProvider(rawValue: provider)
    }
}

/// API response for voices endpoint
struct VoicesResponse: Codable {
    let success: Bool
    let total: Int
    let voices: [Voice]
    let grouped: [String: [Voice]]?
}

/// API response for voice pairs endpoint
struct VoicePairsResponse: Codable {
    let success: Bool
    let pairs: [VoicePair]
}

/// API response for providers endpoint
struct ProvidersResponse: Codable {
    let success: Bool
    let providers: [ProviderInfo]
}

struct ProviderInfo: Codable, Identifiable {
    let id: String
    let name: String
    let description: String
    let available: Bool
    let configured: Bool
    let voiceCount: Int
    let features: [String]

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case description
        case available
        case configured
        case voiceCount = "voice_count"
        case features
    }
}

/// Current voice configuration
struct VoiceConfiguration: Codable {
    var provider: TTSProvider
    var host1VoiceId: String
    var host2VoiceId: String

    static let `default` = VoiceConfiguration(
        provider: .openai,
        host1VoiceId: "openai:nova",
        host2VoiceId: "openai:onyx"
    )
}
